WordPress/gutenberg
# Spacing presets: add editor UI support

## What?
Add UI into the padding toolspanel control to allow for either a custom value to be entered or a [preset spacing value](https://github.com/WordPress/gutenberg/pull/41527) to be selected

## Why?
There needs to be an option for users and theme authors to specify a preset spacing size instead of having to enter a custom value every time

## How?
Adds a new spacing sizes control to the editor components that can be shared between padding, margins, blockgap.

Currently this control is only applied to the padding settings.

## Testing Instructions

- Add a group block and in the block settings panel try setting a range of combinations of values, eg. single-side custom and preset, multi-side custom and prest and switching between them
- Check that any changes are reflected in the editor canvas and frontend
- In core or theme `theme.json` increase `settings.spacing.spacingScale.steps` to a value > 7 and check that segmented range is replaced with a select list
- Note that switching from a preset to custom should copy the preset value into the custom box, but going from custom to preset makes no attempt to transfer the value as the chances of a match with a preset are minimal
- Also check the padding settings in the Site Editor global styles Layout and Block styles

## Screenshots or screencast 

https://user-images.githubusercontent.com/3629020/180367734-83585cd3-2ddb-4736-87df-392121d5f4e4.mp4
